node/network/approval-distribution/src/lib.rs

```diff
@@ -722,9 +726,18 @@ impl State {
 		assignment: IndirectAssignmentCert,
 		claimed_candidate_index: CandidateIndex,
 		rng: &mut R,
+		tracing_label: &'static str,
```

Comment:
I don't think we need to clutter with so many span names, unless we really want to reconstruct the flow which would be anyway hard to do. What makes sense is to differentiate between any difference in processing assignments/approvals and for this we just need two labels: `import-(peer|local)-assignment` and `import-(peer|local)-approval`. When we import something from a peer we always import in approval voting, otherwise that is skipped, since the assignment/approval is local and originates from approval voting subsystem.

Reply:
Would be interesting to compare spans for `MessageSource::Local` vs `MessageSource::Peer(peer_id)` as it should give the book keeping overhead for both approvals and assignments.

Reply:
Based on what @sandreim is saying above, then as the `ApprovalDistributionMessage::DistributeAssignment` is the equivalent of `Local` as it does indeede stem from `approval-voting`, then I'd suggest adding the `local` labels to those spans, and the `peer` label to the spans associated with the `PendingMessage`, and potentially even adding the `peer-id` tag.